packages/legend-graph/src/graph-manager/protocol/pure/v1/V1_PureGraphManager.ts

```diff
@@ -2884,7 +2875,24 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
     );
     return prunedGraphData;
   };
-
+  private createBulkServiceRegistrationInput = (
+    graph: PureModel,
+    services: Service[],
+  ): ServiceRegistrationInput[] => {
+    const graphData = this.getFullGraphModelData(graph);
+    const results: ServiceRegistrationInput[] = [];
+    services.forEach((service) => {
```

Comment:
This was added differently but for any pruning of the graph please use 
`prunePureModelContextData` with a filtering function for the service. See `createBulkServiceRegistrationInput` for an example. 
We need to use this function because of the way dependencies are stored in the graph now to prevent serialization every time.